06_neighborhoods/Transportation/transit_trips_cost_county_subgroups.R

```diff
@@ -0,0 +1,645 @@
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+trips_data_2019 <- trips_data_2019 %>%
+  mutate(
+    race_category = case_when(
+      is.na(perc_white) | is.na(perc_POC) ~ as.character(NA), 
+      perc_white >= 0.6 ~ "Majority White-NH Tracts",
+      perc_POC >= 0.6 ~ "Majority Non-White Tracts",
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+############################################################################
+############################################################################
+############################################################################
+##########################################################################
+###################################################################
+
+# 3a. Collapse to the county level to create ALL values (to be appended later)
+
+cost_all_15 <- cost_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+cost_all_19 <- cost_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_15 <- trips_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_19 <- trips_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+# Now make sure to change the trips to national percentile ranks
+trips_all_15$rank <- rank(trips_all_15$index_transit_trips)
+trips_all_19$rank <- rank(trips_all_19$index_transit_trips)
+# Calculate percentile ranks
+trips_all_15$percentile_rank <- (trips_all_15$rank - 1) / (nrow(trips_all_15) - 1) * 100
+trips_all_19$percentile_rank <- (trips_all_19$rank - 1) / (nrow(trips_all_19) - 1) * 100
+# round to 2 decimals
+trips_all_15$index_transit_trips <- round(trips_all_15$percentile_rank, 2)
+trips_all_19$index_transit_trips <- round(trips_all_19$percentile_rank, 2)
+# keep only what we need
+trips_all_15 <- trips_all_15 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+trips_all_19 <- trips_all_19 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+
+
+# 3b. Collapse accordingly -- to counties and race categories, weighting the measure by HH count per tract
+
+cost_by_race_15 <- cost_data_2015 %>%
```

Comment:
A lot of NAs are created here. I'm not sure if that is supposed to happen.

Reply:
Added in a check for this -- looks like it's OK

Reply:
This still seems conceptually weird to me. NAs are created in situations where a tract essentially had no population. So a county that has all populated tracts won't have any NA. But a county with some non-populated tracts would. I don't quite know what the NA does on cost_by_race_yy. 

Maybe it doesn't matter since they get dropped off when you merge on the county expander.

Reply:
I am going to resolve this one because I think it's OK once you merge on the expander.